File: admin_service/src/types/board.ts
export interface Board {
  id: number;
  name: string;
  description: string;
  category: string;
  categoryId?: number;
  categoryName?: string;
  isActive: boolean;
  displayOrder: number;
  allowAnonymous: boolean;
  requireApproval: boolean;
  postCount: number;
  lastPostAt: string | null;
  createdByUsername: string;
  createdAt: string;
  updatedAt: string;
}

export interface BoardCreateRequest {
  name: string;
  description: string;
  category: string;
  categoryId?: number;
  displayOrder: number;
  allowAnonymous: boolean;
  requireApproval: boolean;
}

export interface BoardUpdateRequest {
  name: string;
  description: string;
  category: string;
  categoryId?: number;
  displayOrder: number;
  allowAnonymous: boolean;
  requireApproval: boolean;
}

export interface BoardApiResponse {
  success: boolean;
  data?: Board | Board[];
  message: string;
}

export const BOARD_CATEGORIES = [
  // Lab실 
  { value: 'Best', label: 'Best', category: 'Lab실' },
  { value: '자유Lab', label: '자유Lab', category: 'Lab실' },
  { value: '정치Lab', label: '정치Lab', category: 'Lab실' },
  { value: '경제Lab', label: '경제Lab', category: 'Lab실' },
  { value: '사회Lab', label: '사회Lab', category: 'Lab실' },
  
  // 취미
  { value: '골프', label: '골프', category: '취미' },
  { value: '낚시', label: '낚시', category: '취미' },
  { value: '축구', label: '축구', category: '취미' },
  { value: '야구', label: '야구', category: '취미' },
  { value: '등산', label: '등산', category: '취미' },
  { value: '건강/헬스', label: '건강/헬스', category: '취미' },
  { value: '여행', label: '여행', category: '취미' },
  { value: '게임', label: '게임', category: '취미' },
  { value: '영화', label: '영화', category: '취미' },
  
  // 먹고살기
  { value: '주식(국장)', label: '주식(국장)', category: '먹고살기' },
  { value: '주식(미장/기타)', label: '주식(미장/기타)', category: '먹고살기' },
  { value: '가상화폐', label: '가상화폐', category: '먹고살기' },
  { value: '자동차', label: '자동차', category: '먹고살기' },
  { value: '부동산', label: '부동산', category: '먹고살기' },
  { value: '육아', label: '육아', category: '먹고살기' },
  
  // 갤러리
  { value: '그림', label: '그림', category: '갤러리' },
  { value: '사진', label: '사진', category: '갤러리' },
  { value: '바탕화면', label: '바탕화면', category: '갤러리' },
  { value: 'AI이미지', label: 'AI이미지', category: '갤러리' },
  { value: '애들은가라(성인)', label: '애들은가라(성인)', category: '갤러리' }
] as const;

export interface BoardCategory {
  id: number;
  name: string;
  description: string;
  displayOrder: number;
  isActive: boolean;
  boardCount: number;
  createdAt: string;
  updatedAt: string;
}

File: backend_service/src/main/java/com/factlab/board/entity/Board.java
package com.factlab.board.entity;

import com.factlab.admin.entity.AdminUser;
import jakarta.persistence.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import com.fasterxml.jackson.annotation.JsonIgnore;

import java.time.LocalDateTime;

@Entity
@Table(name = "boards")
public class Board {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, unique = true, length = 100)
    private String name;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(nullable = false, length = 50)
    private String category;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    @JsonIgnore
    private BoardCategory boardCategory;
    
    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;
    
    @Column(name = "display_order", nullable = false)
    private Integer displayOrder = 0;
    
    @Column(name = "allow_anonymous", nullable = false)
    private Boolean allowAnonymous = false;
    
    @Column(name = "require_approval", nullable = false)
    private Boolean requireApproval = false;
    
    @Column(name = "post_count", nullable = false)
    private Integer postCount = 0;
    
    @Column(name = "last_post_at")
    private LocalDateTime lastPostAt;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private AdminUser createdBy;
    
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
    
    // Constructors
    public Board() {}
    
    public Board(String name, String description, String category) {
        this.name = name;
        this.description = description;
        this.category = category;
    }
    
    // Getters and Setters
    public Long getId() {
        return id;
    }
    
    public void setId(Long id) {
        this.id = id;
    }
    
    public String getName() {
        return name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public String getDescription() {
        return description;
    }
    
    public void setDescription(String description) {
        this.description = description;
    }
    
    public String getCategory() {
        return category;
    }
    
    public void setCategory(String category) {
        this.category = category;
    }
    
    public Boolean getIsActive() {
        return isActive;
    }
    
    public void setIsActive(Boolean isActive) {
        this.isActive = isActive;
    }
    
    public Integer getDisplayOrder() {
        return displayOrder;
    }
    
    public void setDisplayOrder(Integer displayOrder) {
        this.displayOrder = displayOrder;
    }
    
    public Boolean getAllowAnonymous() {
        return allowAnonymous;
    }
    
    public void setAllowAnonymous(Boolean allowAnonymous) {
        this.allowAnonymous = allowAnonymous;
    }
    
    public Boolean getRequireApproval() {
        return requireApproval;
    }
    
    public void setRequireApproval(Boolean requireApproval) {
        this.requireApproval = requireApproval;
    }
    
    public Integer getPostCount() {
        return postCount;
    }
    
    public void setPostCount(Integer postCount) {
        this.postCount = postCount;
    }
    
    public LocalDateTime getLastPostAt() {
        return lastPostAt;
    }
    
    public void setLastPostAt(LocalDateTime lastPostAt) {
        this.lastPostAt = lastPostAt;
    }
    
    public AdminUser getCreatedBy() {
        return createdBy;
    }
    
    public void setCreatedBy(AdminUser createdBy) {
        this.createdBy = createdBy;
    }
    
    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
    
    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
    
    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }
    
    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
    
    public BoardCategory getBoardCategory() {
        return boardCategory;
    }
    
    public void setBoardCategory(BoardCategory boardCategory) {
        this.boardCategory = boardCategory;
    }
}
